### PR TITLE
vsphere: Add support for processors in datastore, host and virtualmachine

### PR DESCRIPTION
**Pull request by bene-dt0**
## Proposed commit message

Adds processor support for the datastore, host and virtual machine metrics data stream within the vSphere integration since it is missing for this integration.

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Author's Checklist

- Availability of Processor fields under datastore, host, and virtualmachine within the vSphere Integration

### CODE REVIEW

**Comment by bene-dt0** [comment]
Hey. Are there any updates on this PR? Is there a timeframe available? Thanks

**Comment by ishleenk17** [comment]
> Hey. Are there any updates on this PR? Is there a timeframe available? Thanks



> Hey. Are there any updates on this PR? Is there a timeframe available? Thanks

The changes look good. Triggered CI. Once that passes, we are GTG.

File: packages/vsphere/changelog.yml [hunk 1]
```diff
@@ -1,4 +1,9 @@
 # newer versions go on top
+- version: "1.14.0"
+  changes:
+    - description: Add processor support for vSphere datastore, host and virtualmachine.
```
Comment (shmsr) [inline packages/vsphere/changelog.yml:4]:
```suggestion
    - description: Add support for processors in datastore, host and virtualmachine.
```

nit; to match with title.

**Comment by ishleenk17** [comment]
We are going to have some big enhancements added to vSphere Integration.
So would be merging this community contribution  before those go in.

**Review by ishleenk17 — APPROVED** [review]

**Pull request merged**